src/Controller/CallbackAction.php

```diff
@@ -59,6 +61,8 @@ public function __invoke(Request $request): Response
             $this->userProfileStorage->setUserProfile($userProfile);
         }
 
+        $this->fetchOrCreateAppCatalogHandler->execute($accessToken);
```

Comment:
as you add the access token to the storage, you don't need to pass it to the `execute()`. Instead, you can inject the `AccessTokenStorage` service into the handler.

Reply:
a bit awkward to have this temporal dependency TBH.